src/tests/ftest/osa/osa_offline_reintegration.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+import time
+import random
+import ctypes
+from apricot import TestWithServers
+from test_utils_pool import TestPool
+from pydaos.raw import (DaosContainer, IORequest,
+                        DaosObj)
+
+
+class OSAOfflineReintegration(TestWithServers):
+    # pylint: disable=too-many-ancestors
+    """
+    Test Class Description: This test runs
+    daos_server offline reintegration test cases.
+
+    :avocado: recursive
+    """
+    def setUp(self):
+        """Set up for test case."""
+        super(OSAOfflineReintegration, self).setUp()
+        self.dmg_command = self.get_dmg_command()
+        self.no_of_dkeys = self.params.get("no_of_dkeys", '/run/dkeys/*')[0]
+        self.no_of_akeys = self.params.get("no_of_akeys", '/run/akeys/*')[0]
+        self.record_length = self.params.get("length", '/run/record/*')
+
+    def get_pool_leader(self):
+        """Get the pool leader
+           Args: None
+           Returns : pool_leader (int)
+        """
+        out = []
+        kwargs = {"pool": self.pool.uuid}
+        out = self.dmg_command.get_output("pool_query", **kwargs)
+        return int(out[0][3])
+
+    def get_pool_version(self):
+        """Get the pool version
+           Args: None
+           Returns : pool_version (int)
+        """
+        out = []
+        kwargs = {"pool": self.pool.uuid}
+        out = self.dmg_command.get_output("pool_query", **kwargs)
```

Comment:
Failure of the dmg command should be handled. I would suggest adding `@fail_on(CommandFailure)` as a decorator to the function. You'd have to also add this import to the test: `from avocado import fail_on`

Reply:
Done.